Nef_3/include/CGAL/Nef_3/SNC_external_structure.h

```diff
@@ -255,34 +255,50 @@ class SNC_external_structure_base : public SNC_decorator<SNC_structure_>
   typedef CGAL::Unique_hash_map<SFace_const_handle,bool> SFace_visited_hash;
   typedef CGAL::Unique_hash_map<SFace_const_handle,bool> Shell_closed_hash;
 
+  typedef std::vector<SFace_handle> SFace_list;
+  typedef std::vector<Halffacet_handle> Halffacet_list;
+
   using SNC_decorator::visit_shell_objects;
+  using SNC_decorator::store_boundary_object;
+  using SNC_decorator::set_volume;
   using SNC_decorator::link_as_inner_shell;
   using SNC_decorator::link_as_outer_shell;
   using SNC_decorator::link_as_prev_next_pair;
   using SNC_decorator::get_visible_facet;
   using SNC_decorator::adjacent_sface;
   using SNC_decorator::make_twins;
 
+  struct Shell_entry {
+    Shell_entry(SFace_handle f) : minimal_sface(f) {}
+    SFace_handle minimal_sface;
+    SFace_list connected_sfaces;
+    Halffacet_list connected_halffacets;
+  };
+
+  typedef std::vector<Shell_entry> Shell_entry_list;
+
   struct Shell_explorer {
-    const SNC_decorator& D;
-    Sface_shell_hash&  ShellSf;
-    Face_shell_hash&   ShellF;
-    //    Shell_closed_hash& Closed;
+    Sface_shell_hash&   shell_lookup;
```

Comment:
Hello @GilesBathgate, can you please the next time avoid to put a simple renaming in the same commit. This makes it harder to see the *real*  improvement.

Reply:
Yes in future I will do that. I am not sure why I didn't this time 🤷